File: app/src/main/java/com/polstech/kotlinsamples/_2_functions/_2_DefaultFuncArgs.java
package com.polstech.kotlinsamples._2_functions;

/**
 * Created by polprashant on 16/11/17.
 */

public class _2_DefaultFuncArgs {
    public void find(int[] array, int firstElement, int secondElement, int searchItem) {
    }

    public void find(int[] array, int searchItem) {
        find(array, 0, array.length, searchItem);
    }

    public void find(int[] array, int firstElement, int searchItem) {
        find(array, firstElement, array.length, searchItem);
    }


}

File: app/src/main/kotlin/_2_functions/_2_DefaultFuncArgs.kt
package _2_functions

/**
 * Created by prashantspol on 11/9/2017.
 */

class _2_DefaultFuncArgs {

    @JvmOverloads
    fun find(array:IntArray,
             startIndex: Int = 0,
             endIndex: Int = array.size - 1,
             searchItem: Int = 1) {


        // implementation for search
    }

    fun call() {
        var arr = intArrayOf()
        find(arr)

        find(array= arr, searchItem = 1, endIndex = 6, startIndex = 0)
    }
}